test/parallel/test-fs-readfile-tostring-fail.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const kMaxLength = process.binding('buffer').kMaxLength;
+
+common.refreshTmpDir();
+
+const file = path.join(common.tmpDir, 'toobig.txt');
+const stream = fs.createWriteStream(file, {
+  flags: 'a'
+});
+
+const size = 269 * 1024 * 1024;
+
+const a = new Array(1000000).join('a').toString();
+const len = kMaxLength / 4;
+
+for (var i = 0; i < size; i += 1000000) {
+  stream.write(a);
+}
+
+stream.end();
+stream.on('finish', common.mustCall(function() {
+  // make sure that the toString does not throw an error
+  fs.readFile(file, 'utf8', common.mustCall(function(err, buf) {
+    assert.strictEqual('toString failed', err.message);
```

Comment:
This test will be flaky on the pi's since there's higher likelihood that they'll run out of memory.